packages/electron-builder/src/publish/PublishManager.ts

```diff
@@ -158,8 +158,12 @@ export class PublishManager implements PublishContext {
     }
   }
 
-  private getOrCreatePublisher(publishConfig: PublishConfiguration, platformPackager: PlatformPackager<any>): Publisher | null {
-    const providerCacheKey = `${publishConfig.provider}-${platformPackager.platform.name}`
+  private getOrCreatePublisher(publishConfig: PublishConfiguration, platformPackager: PlatformPackager<any>, target: Target | null): Publisher | null {
+    let providerCacheKey = `${publishConfig.provider}-${platformPackager.platform.name}`
+    if (target) {
```

Comment:
The conditional is only needed because `target` might be `null` (not sure under what circumstances that could happen)